json-smart/src/test/java/net/minidev/json/test/TestOverflow.java

```diff
@@ -28,7 +30,27 @@ public void stressTest() throws Exception {
 			assertEquals(e.getErrorType(), ParseException.ERROR_UNEXPECTED_JSON_DEPTH);
 			return;
 		}
-		assertTrue(false);
+		fail();
+	}
+
+	@Test
+	public void shouldNotFailWhenInfiniteJsonDepth() throws Exception {
+		int size = 500;
+		StringBuilder sb = new StringBuilder(10 + size*4);
+		for (int i=0; i < size; i++) {
+			sb.append("{a:");
+		}
+		sb.append("true");
+		for (int i=0; i < size; i++) {
+			sb.append("}");
+		}
+		String s = sb.toString();
+		try {
+			JSONParser parser = new JSONParser(DEFAULT_PERMISSIVE_MODE & ~JSONParser.FINITE_JSON_DEPTH);
```

Comment:
why `FINITE_JSON_DEPTH` could not benefit of an import static wherever `DEFAULT_PERMISSIVE_MODE` can.

Reply:
`DEFAULT_PERMISSIVE_MODE & ~FINITE_JSON_DEPTH` ?
should be:
`DEFAULT_PERMISSIVE_MODE | FINITE_JSON_DEPTH` ?

Reply:
I do not remember this code, but I think that the default parser value is 0
...
So the issue is that the flag should be call UNLIMITED_JSON_DEPTH and do the opposite.
The limit is enabled by default for now, and can be disabled in some rare cases.

Reply:
to the fixed version should be: `DEFAULT_PERMISSIVE_MODE | UNLIMITED_JSON_DEPTH`